RadixWalletTests/ProfileTests/Utils/ProfileBuilder.swift

```diff
@@ -0,0 +1,101 @@
+import Foundation
+@testable import Radix_Wallet_Dev
+
+// MARK: - ProfileBuilder
+public final class ProfileBuilder {
+	private let profileID: ProfileSnapshot.Header.ID
+	private var bdfs: PrivateHDFactorSource?
+	private var networks: IdentifiedArrayOf<Network> = []
+
+	public init(
+		id: ProfileSnapshot.Header.ID? = nil
+	) {
+		self.profileID = id ?? UUID()
+	}
+}
+
+extension ProfileBuilder {
+	public func build() -> Profile {
+		guard let bdfs else { fatalError("No BDFS, unable to create Profile.") }
+		let deviceInfo = DeviceInfo(description: "Builder", id: UUID(), date: .now)
+		let networksDictionary: OrderedDictionary<NetworkID, Profile.Network> = .init(uniqueKeysWithValues: self.networks.map {
+			(key: $0.id, value: Profile.Network(networkID: $0.id, accounts: .init(rawValue: $0.accounts)!, personas: [], authorizedDapps: []))
+		})
+		assert(networksDictionary.keys.contains(.mainnet))
+		let profile = Profile(
+			header: .init(
+				creatingDevice: deviceInfo,
+				lastUsedOnDevice: deviceInfo,
+				id: self.profileID,
+				lastModified: .now,
+				contentHint: .init(numberOfAccountsOnAllNetworksInTotal: self.networks.reduce(0) { $0 + $1.accounts.count })
+			),
+			deviceFactorSource: bdfs.factorSource,
+			appPreferences: .init(gateways: .init(current: .mainnet)),
+			networks: .init(dictionary: networksDictionary)
+		)
+		assert(!profile.network!.getAccounts().isEmpty)
+		return profile
+	}
+}
+
+// MARK: ProfileBuilder.Network
+extension ProfileBuilder {
+	private final class Network: Identifiable {
+		let id: NetworkID
+		var accounts: IdentifiedArrayOf<Profile.Network.Account>
+		init(id: NetworkID, accounts: IdentifiedArrayOf<Profile.Network.Account> = []) {
+			self.id = id
+			self.accounts = accounts
+		}
+
+		func addAccount(_ account: Profile.Network.Account) {
+			accounts.append(account)
+		}
+	}
+}
+
+extension ProfileBuilder {
+	// TODO: Refactor to use real clients aligning with prod code?
```

Comment:
tried using `Live` clients.. it... did ...not...work :/ I tried many various combinations of `withDependencies` but I could not get a selective `Live` impl of e.g. `FactorSourceClient` but with a `Test` impl of e.g. `SecureStorageClient`. I might give it a go later.